src/main/java/somethingrandom/dataaccess/google/tasks/TaskList.java

```diff
@@ -34,4 +38,21 @@ public void add(Item item) throws DataAccessException, IOException {
         request.put("notes" , item.toString());
         provider.request(new APIRequestBody.JSONBody("POST", request), "https://tasks.googleapis.com/tasks/v1/lists/" + identifier + "/tasks");
     }
+
+    public Collection<Item> getAll() throws AuthenticationException, IOException {
+        JSONObject response = provider.request(new APIRequestBody.JSONBody("GET", new JSONObject()), "https://tasks.googleapis.com/tasks/v1/lists/" + identifier + "/tasks");
+        JSONObject items = response.getJSONObject("items");
+        Collection<Item> allItems = new ArrayList<>();
+        for (Object item: response.getJSONArray("items")) {
+            JSONObject jsonItem = new JSONObject(item);
```

Comment:
Should've pointed this out the first time: I don't think this is right. The [documentation](https://stleary.github.io/JSON-java/org/json/JSONObject.html) says:

> [JSONObject](https://stleary.github.io/JSON-java/org/json/JSONObject.html#JSONObject-java.lang.Object-)([Object](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true) bean)
Construct a JSONObject from an Object using bean getters.

But JSONObject isn't a bean. Probably `(JSONObject) item` would work?